src/compiler/xslt/compile/compile-scenario.xsl

```diff
@@ -45,11 +45,11 @@
             </xsl:call-template>
          </xsl:message>
       </xsl:if>
-      <xsl:if test="$context and $call/@function">
+      <xsl:if test="$context and $call/@function and not($is-external)">
          <xsl:message terminate="yes">
             <xsl:call-template name="x:prefix-diag-message">
                <xsl:with-param name="message" as="xs:string">
-                  <xsl:text>Can't set a context and call a function at the same time</xsl:text>
+                  <xsl:text expand-text="yes">Setting a context and calling a function at the same time is supported only when /{$initial-document/x:description => name()} has @run-as='external'.</xsl:text>
```

Comment:
A suggestion for streamlining the message: "Setting a context for calling a function is supported only..."

Reply:
Thanks, @galtm. b445bf9642e62e7e461ea52f81d6e3c74f7b9791 integrates it.

I'm merging this.